### PR TITLE
GitHub scope CTA global banner

### PR DESCRIPTION
**Pull request by artemruts**
### Description

These changes add a global dismissible banner for user that is:
* on `.com`
* enrolled in private code
* is missing "private" scope for GitHub

### Screenshot
Screen Shot 2021-06-03 at 10 56 44 PM

### Questions

1. When we display a global banner do we want to hide a similar alert on the "code host connections" page?
2. We'll make requests for `grantedScopes` only if the user does a full page refresh (unmount the top level app),
when user navigates between routes we don't make additional requests. Is it ok?
3. Doesn't [this grantedScopes resolver code](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@ar/global-cta/-/blob/cmd/frontend/graphqlbackend/external_service.go?utm_source=VSCode-1.3.0#L181) mean we'll show a global alert and an update button for GitHub code host if we can't resolve the scope or had and error?

### CODE REVIEW

**Comment by ryanslade** [comment]
Regarding the resolver code:

Yes, you're right since it'll just return zero scopes. I had to add this workaround since when we request a list of external services we include the scopes. Unfortunately if any resolver returns an error the entire request fails. 

I saw this yesterday when there was one misconfigured external service. When visiting the site admin external service list page the entire request failed because the single service was unable to fetch its scopes. 

Maybe a better solution here is to *only* request the scopes when we know we need them and not include it on all external service queries. I wasn't sure how to do this as it looks like we just have a single entrypoint for doing requests on the frontend.

File: client/web/src/SourcegraphWebApp.tsx [hunk 1]
```diff
@@ -210,6 +211,12 @@ interface SourcegraphWebAppState extends SettingsCascadeProps {
      */
     enableCodeMonitoring: boolean
 
+    /**
+     * Whether the user is missing a private scope for GitHub external service
+     * and needs to re-add the external service.
```
Comment (ryanslade) [inline client/web/src/SourcegraphWebApp.tsx:216]: I guess technically they're not adding a new service, but requesting a new token for an existing service.

Comment (artemruts) [reply client/web/src/SourcegraphWebApp.tsx]: Thank you!

**Comment by artemruts** [comment]
@ryanslade closing in favor of https://github.com/sourcegraph/sourcegraph/pull/21950

**Pull request closed without merging**